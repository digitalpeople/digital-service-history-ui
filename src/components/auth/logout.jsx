import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import { Redirect } from 'react-router-dom';
import { logoutAction } from '../../store/auth/auth.action';

export const Logout = (props) => {
  const { onLogout } = props;

  onLogout();

  return <Redirect to="/login" />;
};

Logout.propTypes = {
  onLogout: PropTypes.func.isRequired,
};

const mapDispatchToProps = dispatch => ({
  onLogout: () => dispatch(logoutAction()),
});

export default connect(null, mapDispatchToProps)(Logout);
