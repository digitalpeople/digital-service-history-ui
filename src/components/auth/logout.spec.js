import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Logout } from './logout';

configure({ adapter: new Adapter() });

describe('Logout', () => {
  it('should render component', () => {
    // ASSIGN
    const props = {
      onLogout: jest.fn(),
    };

    // ACT
    const actual = shallow(<Logout {...props} />);

    // ASSERT
    expect(actual.text()).toMatchSnapshot();
    expect(props.onLogout).toHaveBeenCalled();
  });
});
