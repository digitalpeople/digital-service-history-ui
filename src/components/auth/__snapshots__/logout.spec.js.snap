// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Logout should render component 1`] = `"<Redirect />"`;
